### DISCUSSION TITLE
fromEnvironment doesn't pick up the Environment variables

### DISCUSSION THREAD

**Discussion by isaac-gofa** [codemagic-ci-cd/codemagic-docs#1899] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Defined CLOUDFLARE_API_KEY=xxxxx in Environment variables tab in Workflow Editor
2. apiKey = const String.fromEnvironment('CLOUDFLARE_API_KEY',
        defaultValue: "0000");
3. apiKey is still 0000. 
4. I tried also "apiKey = const String.fromEnvironment('\$CLOUDFLARE_API_KEY',
        defaultValue: "0000");"

### Expected results

my secret api key specified in Environment variables tab in Workflow Editor

### Actual results

0000

**Comment by isaac-gofa** [comment, marked as answer]
I have resolved the problem with the following set-up.

environment variables:
APP_CONFIG={"CLOUDFLARE_API_KEY": xxxxx}

pre-build scripts:
# Create directory if it doesn't exist
mkdir -p $CM_BUILD_DIR/config
# Write out the environment variable as a json file
echo $APP_CONFIG > $CM_BUILD_DIR/config/prod.json

build command flags:
--dart-define-from-file config/prod.json

**Comment by ntfnd404** [comment]
How to set APP_CONFIG={"CLOUDFLARE_API_KEY": xxxxxx} in Workflow Editor ? Is APP_CONFIG a variable name on the variables tab? And where do you enter "build command flags:
--dart-define-from-file config/prod.json"
